validator/project_validator.go

```diff
@@ -222,6 +222,60 @@ func getDistrosForProject(ctx context.Context, projectID string) (ids []string,
 	return ids, aliases, nil
 }
 
+func CheckProject(ctx context.Context, project *model.Project, config *model.ProjectConfig, ref *model.ProjectRef, settings *evergreen.Settings, includeLong, projectRefProvided, projectRefErrored bool) ValidationErrors {
```

Comment:
A couple of things here:
1) Exported function needs a function comment
2) We want to avoid having this many variables passed in that have the same type, since its easy to mistake which one is which and pass things in the wrong order. If we need this many variables, we should use different types. 
3) projectRefProvided boolean is a little confusing; would explain how this ties into err and project = nil in the comments
4) We can just call evergreen.GetEnvironment().Settings() in this function instead of passing it in, since we just call that in both places.